setup.py

```diff
@@ -84,8 +102,56 @@ def build_extensions(self):
             self.compiler.compiler_so.remove('-Wstrict-prototypes')
         except (ValueError, AttributeError):
             pass
+
+        cppflags = []
+        if 'CPPFLAGS' in os.environ:
+            cppflags.append(os.environ['CPPFLAGS'])
+        cxxflags = []
+        if 'CXXFLAGS' in os.environ:
+            cxxflags.append(os.environ['CXXFLAGS'])
+        ldflags = []
+        if 'LDFLAGS' in os.environ:
+            ldflags.append(os.environ['LDFLAGS'])
+        env = os.environ.copy()
+
+        if has_flag(self.compiler, '-fvisibility=hidden'):
```

Comment:
May be worth skipping these on Windows?  VS hides symbols by default (https://pybind11.readthedocs.io/en/stable/faq.html#how-can-i-create-smaller-binaries) and subprocess invokes are rather slow there, so avoiding unnecessary subprocesses matters (a bit).

Reply:
Done.